core/src/raw/path_cache.rs

```diff
@@ -0,0 +1,230 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::raw::*;
+use crate::*;
+use async_trait::async_trait;
+use moka::sync::Cache;
+use tokio::sync::{Mutex, MutexGuard};
+
+/// The trait required for path cacher.
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+pub trait PathQuery {
+    /// Fetch the id for the root of the service.
+    async fn root(&self) -> Result<String>;
+    /// Query the id by parent_id and name.
+    async fn query(&self, parent_id: &str, name: &str) -> Result<Option<String>>;
+    /// Create dir by parent_id and name.
+    async fn create_dir(&self, parent_id: &str, name: &str) -> Result<String>;
```

Comment:
What does this mean?

Reply:
What I mean by that is that the operations emphasize that they will only be performed once under that parent_id, and will not be performed recursively, to avoid any misinterpretation that they will be queried or created recursively.